### ISSUE TITLE
Working Notes on Working in a Git Branch

### ISSUE THREAD

**Issue by ebeshero** [ebeshero/Amadis-in-Translation#52]
It's tricky to keep your branch distinctive while pulling in changes from the master. The process seems different from working in a fork. I have created an "Elisa" branch on the Amadis repository, while my collaborators have created forks in their own spaces. My "Elisa" branch is supposed to function like a fork, in that I can use it as a distinctly separate workspace from the remote origin. However, as changes are pushed to the remote origin, I need to keep the "Elisa" branch updated with those changes, _without losing its distinct history._
- First of all, upon cloning my repository, I need to track the "Elisa" branch. (Git won't automatically track it on a clone of the repo.) To track, use:
  `git branch --track Elisa origin/Elisa`
- I think I need to use rebase to keep Elisa in sync with changes from master, except rebase isn't working as I'd expect. See https://www.atlassian.com/git/tutorials/merging-vs-rebasing/conceptual-overview which explains rebase. Rebase would _seem_ to move my branch ahead so it's as if it diverged at the point of the latest commit to the remote origin. However, when I try to first clone the repo, track the Elisa branch, and then rebase the branch with the master (as described in the tutorial), my repository _loses its distinctiveness_. The Elisa branch is supposed to contain a set of files that are not in the master branch. When I rebase and when I merge I think I get the same result: those files are removed to make my branch like the master. (So, what's the solution? Can anyone help?)

@rjp43 @djbpitt @sydb @HelenaSabel

**Comment by RJP43** [comment]
Two questions: when you do the merge or rebase which branch do you have checked out ? Also has the Elisa branch ever been pushed to the remote (server) repository?

**Comment by RJP43** [comment]
*rebase

**Comment by ebeshero** [comment]
@RJP43 I have checked out the Elisa branch before I try a rebase with master. And yes, the Elisa branch exists on the remote repository, and my collaborators could choose to track it and check it out when we are working on experiments with our project schema. (We don't want to store those schema experiments in the main repository because they don't work yet to constrain our project files--so that gives you an idea of what the branch is for!)

**Comment by ebeshero** [comment]
@HelenaSabel I renamed our experimental ODD files in the "Elisa" branch and I think I've successfully updated everything. (The name changes generated a merge conflict when I attempted to rebase the "Elisa" branch with master, but resolving the merge conflict involved nothing more than using `git add .` to track the new files. 

I appear to have our current Schematron loaded into the "Elisa" branch, and I think I should be able to keep rebasing it now, so I can work in here as we intended. 

So all that is to say, all's well with this branch--go and check it out, Helena, and see how it looks to you!
@rjp43
